services/org/org.go

```diff
@@ -53,15 +59,100 @@ func DeleteOrganization(org *organization.Organization) error {
 	path := user_model.UserPath(org.Name)
 
 	if err := util.RemoveAll(path); err != nil {
-		return fmt.Errorf("Failed to RemoveAll %s: %w", path, err)
+		return fmt.Errorf("failed to RemoveAll %s: %w", path, err)
 	}
 
 	if len(org.Avatar) > 0 {
 		avatarPath := org.CustomAvatarRelativePath()
 		if err := storage.Avatars.Delete(avatarPath); err != nil {
-			return fmt.Errorf("Failed to remove %s: %w", avatarPath, err)
+			return fmt.Errorf("failed to remove %s: %w", avatarPath, err)
 		}
 	}
 
 	return nil
 }
+
+// RenameOrganization renames an organization.
+func RenameOrganization(ctx context.Context, org *org_model.Organization, newName string, onlyCapitalization bool) error {
+	if !org.AsUser().IsOrganization() {
+		return fmt.Errorf("cannot rename user")
+	}
+
+	if err := user_model.IsUsableUsername(newName); err != nil {
+		return err
+	}
+
+	ctx, committer, err := db.TxContext(ctx)
+	if err != nil {
+		return err
+	}
+	defer committer.Close()
+
+	oldName := org.Name
+
+	if onlyCapitalization {
+		org.Name = newName
+		if err := user_model.UpdateUserCols(ctx, org.AsUser(), "name"); err != nil {
+			org.Name = oldName
+			return err
+		}
+		if err := committer.Commit(); err != nil {
+			org.Name = oldName
+			return err
+		}
+		return nil
+	}
+
+	isExist, err := user_model.IsUserExist(ctx, org.ID, newName)
+	if err != nil {
+		return err
+	}
+	if isExist {
+		return user_model.ErrUserAlreadyExist{
+			Name: newName,
+		}
+	}
```

Comment:
It looks like `onlyCapitalization` only affects this? If yes, could it be:

```
	if isExist && lower_case_not_match {
		return user_model.ErrUserAlreadyExist{
			Name: newName,
		}
	}
```

Then remove the `onlyCapitalization` argument?

Reply:
I don't understand what's your meaning. I have updated this PR.